kratos/python_scripts/check_scalar_to_nodes_process.py

```diff
@@ -1,144 +1,140 @@
+        if settings.Has("interval"):
+            base_settings.AddValue("interval", settings["interval"])
+        if settings.Has("value"):
+            base_settings.AddValue("value", settings["value"])
+        if settings.Has("tolerance_rank"):
+            base_settings.AddValue("tolerance_rank", settings["tolerance_rank"])
+
+        # Construct the base process
+        super(CheckScalarToNodesProcess, self).__init__(Model, base_settings)
+
+        # Copy from input
+        self.settings = settings
+
+        # Here i do a trick, since i want to allow "value" to be a string or a double value
+        if(self.settings.Has("value")):
+            if(self.settings["value"].IsString()):
+                default_settings["value"].SetString("0.0")
+
+        # Validate and assign
+        self.settings.ValidateAndAssignDefaults(default_settings)
+
+        # Admissible values for local axes, are "empty" or
+        #"local_axes"               :{
+        #    "origin" : [0.0, 0.0, 0.0]
+        #    "axes"  : [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0] ]
+        #    }
+        self.non_trivial_local_system = False
+        if(self.settings["local_axes"].Has("origin")): # not empty
+            self.non_trivial_local_system = True
+            self.R = KratosMultiphysics.Matrix(3,3)
+            self.R[0,0] = self.settings["local_axes"]["axes"][0][0].GetDouble()
+            self.R[0,1] = self.settings["local_axes"]["axes"][0][1].GetDouble()
+            self.R[0,2] = self.settings["local_axes"]["axes"][0][2].GetDouble()
+            self.R[1,0] = self.settings["local_axes"]["axes"][1][0].GetDouble()
+            self.R[1,1] = self.settings["local_axes"]["axes"][1][1].GetDouble()
+            self.R[1,2] = self.settings["local_axes"]["axes"][1][2].GetDouble()
+            self.R[2,0] = self.settings["local_axes"]["axes"][2][0].GetDouble()
+            self.R[2,1] = self.settings["local_axes"]["axes"][2][1].GetDouble()
+            self.R[2,2] = self.settings["local_axes"]["axes"][2][2].GetDouble()
+
+            self.x0 = KratosMultiphysics.Vector(3)
+            self.x0[0] = self.settings["local_axes"]["origin"][0].GetDouble()
+            self.x0[1] = self.settings["local_axes"]["origin"][1].GetDouble()
+            self.x0[2] = self.settings["local_axes"]["origin"][2].GetDouble()
+
+        # Getting the mesh
+        self.mesh = self.model_part.GetMesh(self.settings["mesh_id"].GetInt())
+
+        # Getting reference configuration
+        self.reference_configuration = self.settings["reference_conf"].GetBool()
+
+    def ExecuteFinalizeSolutionStep(self):
+        """ This method is executed in order to finalize the current step
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        process_info = self.model_part.ProcessInfo
+        current_time = process_info[KratosMultiphysics.TIME] - process_info[KratosMultiphysics.DELTA_TIME]
```

Comment:
?

Reply:
Current implementation, I will not touch it

Reply:
strange ...